Automattic/jetpack
# Carousel: info and comment sections stay open once toggled on

#### Changes proposed in this Pull Request:

* When a slide is advanced, this PR will stop hiding the photo info and comments section if they were toggled on, and scroll the carousel container to the top if needed.
* Having to repeatedly click the info/comment toggle for each slide can be taxing/annoying.
  * This somewhat address user requests of showing items outside of the info section (e.g. the full description, full size link, etc.), as those will be a simple scroll down once the info section is toggled on once (rather than hitting the toggle for each slide).

#### Jetpack product discussion

None/here.

An option to always have the info section opened by default was mentioned in #20770

#### Does this pull request change what data or activity we track or use?

No.

#### Testing instructions:

* With these changes pulled down, and the carousel feature enabled, open/create a gallery on your site.
* It will help if one of the images has a long description for testing.
* When viewing that image in the carousel, click the `i` info icon to expand the info section and scroll down.
* Without scrolling to the very top, advance the slide.
* The next slide should scroll to the top.
* The info section should still be toggled on. Similar behavior should exist for the comment section if that is toggled on instead of the info section.
* Check on desktop with a variety of resolutions and browsers.
* Especially, check on real mobile devices, quickly swiping to advance slides while the info/comment section is toggled open to see if any conflicts occur.
